### DISCUSSION TITLE
PT100 3 wires

### DISCUSSION THREAD

**Discussion by tukutt** [nebhead/PiFire#134] [Ideas]
Hi,

Congrats again for your project. 

I see some Pt100 / Pt1000 have 3 wires (GMG seems to use this type of probes). Do you think the accuracy is much better?

![image](https://user-images.githubusercontent.com/9823334/195857730-3707dcfb-00a5-425b-b907-cfea2f2f6ecb.png)

A MAX31865 is necessary for each probe. I don't know if it's possible to connect 3 or 4 MAX31865.

https://learn.adafruit.com/adafruit-max31865-rtd-pt100-amplifier?view=all

Other idea, GMG have an ambient probe but I don't know why !

+

**Comment by nebhead** [comment]
> Congrats again for your project.

Thank you! 
 
> I see some Pt100 / Pt1000 have 3 wires (GMG seems to use this type of probes). Do you think the accuracy is much better?

So, yes, I think the accuracy is much better with these types of probes and with the MAX31865.  And, PiFire does currently support the MAX31865, for the grill probe only.  The reason for supporting only a single probe, is due to the fact that this controller support SPI for interfacing to the Raspberry Pi, and it takes up one of the two available SPI interfaces on the system.  To add enough of these devices to support all the probes, we would need to add some additional SPI multiplexing hardware to the project which is expensive and complex.  

However, I would argue that the accuracy for any of these probes does not need to be so good for this particular project.  We we can get away with having less accuracy and speed, because we don't need them to be more accurate than a fraction of a degree in temperature (C or F).  In other industrial or scientific applications, where accuracy down to the tiniest of degree changes is important, the MAX31865 and these type of RTD probes is more appropriate.  

With all of that said, my personal belief is that it's not really all that necessary.  That's why you might see some other projects just using ADC's (Heatermeter, etc.), like I've chosen to go with here in PiFire.  

> A MAX31865 is necessary for each probe. I don't know if it's possible to connect 3 or 4 MAX31865.

Yes, but this would require another device to multiplex the SPI bus, and modification of the code.  Also, this may require more tight tolerances for the routing of these devices, which will increase cost and complexity.  

> Other idea, GMG have an ambient probe but I don't know why !

So, this is not a bad idea and has actually been implemented by @weberbox in this project.  With his PCB design, you can utilize the fourth ADC input for a second Grill Probe, which can be used to average the temperatures.  This can be useful if you have a large cook chamber and have a wider variation of temperature across the entire chamber.

**Reply by tukutt** [reply]
Thanks for your reply. It's overkill to use 3 wires probes !
